server-spi-private/src/main/java/org/keycloak/organization/OrganizationProvider.java

```diff
@@ -29,11 +30,15 @@ public interface OrganizationProvider extends Provider {
      * The internal ID of the organization will be created automatically.
      * @param realm Realm owning this organization.
      * @param name String name of the organization.
-     * @throws ModelDuplicateException If there is already an organization with the given name 
+     * @throws ModelDuplicateException If there is already an organization with the given name
      * @return Model of the created organization.
      */
     OrganizationModel createOrganization(RealmModel realm, String name);
 
+    void updateOrganization(RealmModel realm, OrganizationModel organization);
+
+    OrganizationModel getOrganization(String id);
+
```

Comment:
IMO `updateOrganization` method is not needed as Keycloak is designed in a way that all models propagates changes to its underlying entity automatically (a.k.a all jpa entities are managed entities). So it should be enough to just update the model. 

```suggestion
    OrganizationModel getOrganizationById(RealmModel realm, String id);
```